### ISSUE TITLE
Piping the result to `head` or something leads to `BrokenPipeError: [Errno 32] Broken pipe`

### ISSUE THREAD

**Issue by gh640** [gh640/gh-repo-list#9]
```bash
❯ gh repo-list --type=starred | head
```

```text
Traceback (most recent call last):
  File "/Users/<user>/.local/share/gh/extensions/gh-repo-list/gh-repo-list", line 162, in <module>
    main()
  File "/Users/<user>/.local/share/gh/extensions/gh-repo-list/gh-repo-list", line 24, in main
    print(output)
BrokenPipeError: [Errno 32] Broken pipe
Exception ignored in: <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
BrokenPipeError: [Errno 32] Broken pipe
```

PR:

- #8

**Comment by gh640** [comment]
This was resolved in #8.

**Comment by gh640** [comment]
Ref:

- [signal — Set handlers for asynchronous events — Python 3.12.4 documentation](https://docs.python.org/3/library/signal.html#note-on-sigpipe)

```py
import os
import sys

def main():
    try:
        # simulate large output (your code replaces this loop)
        for x in range(10000):
            print("y")
        # flush output here to force SIGPIPE to be triggered
        # while inside this try block.
        sys.stdout.flush()
    except BrokenPipeError:
        # Python flushes standard streams on exit; redirect remaining output
        # to devnull to avoid another BrokenPipeError at shutdown
        devnull = os.open(os.devnull, os.O_WRONLY)
        os.dup2(devnull, sys.stdout.fileno())
        sys.exit(1)  # Python exits with error code 1 on EPIPE

if __name__ == '__main__':
    main()
```
